### PR TITLE
Initial support for Pico 2 (squashed version of PR 77368)

### PR DESCRIPTION
**Pull request by kartben**
In the interest of getting the work from PR #77368 finally merged into upstream main, this PR proposes the **exact same change set** with the only difference being:

The following commits have been squashed together. As per our [contribution guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#dco-sign-off), sign-off entry of the original author has been kept intact, and I've added mine as well:

SoC:

- Relevant part of "rp2350: Define and implement a cpucluster of Cortex-M33s" squashed into: 
- soc: rp2350: Add initial support for the Raspberry Pi RP2350 

Board:

- Relevant part of "rp2350: Define and implement a cpucluster of Cortex-M33s" and 
- "doc: rpi_pico2: Document Raspberry Pi Pico 2 and related changes" squashed into:
- "boards: Add initial support for the Raspberry Pi Pico 2"

### CODE REVIEW

File: soc/raspberrypi/rpi_pico/rp2350/Kconfig
Comment (nordicjm) [inline soc/raspberrypi/rpi_pico/rp2350/Kconfig:12]: seems to be wrong commit, should be in previous soc commit

Comment (kartben) [reply soc/raspberrypi/rpi_pico/rp2350/Kconfig:12]: @nordicjm better now?

Comment (nordicjm) [reply soc/raspberrypi/rpi_pico/rp2350/Kconfig:12]: Perfect! Thanks

**Review by nordicjm — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Review by soburi — APPROVED** [review]

**Comment by ajf58** [comment]
[5c39bb2](https://github.com/zephyrproject-rtos/zephyr/commit/5c39bb22a0caff1ace4d360a9e0f2f7066c55dfc) looks like it's both authored by and committed by kartben. 

This, maintainers (@soburi @carlescufi @nordicjm), is why I really really really tried so many times to discourage futzing around and rebasing code. It's such an easy mistake to fall into exactly this, where as far as git's history will record (and gitlens et al will show when people are reviewing this), all of the work in that commit was done by @kartben . 

It's really late in the evening in my timezone. I understand and am grateful kartben has put the effort it and actually did something in front of a keyboard., but man am I pretty blue that the finale of this was a rush-job approval that ended up with this artefact.

**Comment by soburi** [comment]
That's why the repeated requests were made "in good faith of reviewers."

It was clear from the beginning that the terms you agreed to when you committed this code allowed you to do this. As explained before, editing by committers is a common practice in OSS, including Linux development, and there is nothing to prevent this.

When the vote result was announced, you had no choice but to fix it yourself or accept it if someone else fixes it. However, by not following the instructions of the vote, you refused to reach a consensus. And by refusing even yesterday's update by the maintainer, you also refused the last chance to merge in your name.

Escalation is provided for when consensus cannot be reached, and refusing to accept the result of the vote can only be interpreted as a refusal to reach a consensus with this project and the community.

If consensus cannot be reached, it is clear that the only option is to proceed based on the terms and written agreement. That is what the terms and agreement are for.

**Comment by ajf58** [comment]
> the last chance to merge in your name.

@soburi , that's not what I'm talking about.  I am talking about the difference between a committer and an author within `git`. Please take this as constructive feedback: quite often when I try and raise a question about one topic you confidently reply about something else, often at great length. This makes it quite challenging to move the conversation forward. 

Please go and look at https://github.com/zephyrproject-rtos/zephyr/commit/5c39bb22a0caff1ace4d360a9e0f2f7066c55dfc . In the process of squashing, both the author (which should be me), and the committer have changed. IMHO former is not correct, the latter is. I really don't think kartben meant to do that: they didn't do it in their previous version of the PR, but post-review feedback they did.

**Comment by kartben** [comment]
Hi @ajf58 

> Please go and look at [5c39bb2](https://github.com/zephyrproject-rtos/zephyr/commit/5c39bb22a0caff1ace4d360a9e0f2f7066c55dfc) . In the process of squashing, both the author (which should be me), and the committer have changed. IMHO former is not correct, the latter is. I really don't think kartben meant to do that: they didn't do it in their previous version of the PR, but post-review feedback they did.

Let me try to explain the thought process here.

You initially asked me to drop your sign-off from the squashed commits as a way to express your discontent regarding how this was in your opinion not needed. So initially, when I pushed to your branch, squashed commit was **"author: Benjamin, sign-off: Benjamin, Co-authored-by: Andrew"**. The fact your sign-off had been dropped was an OK thing to do as per my interpretation of the [DCO](https://docs.zephyrproject.org/latest/contribute/guidelines.html#developer-certification-of-origin-dco), in particular clause (b). However, our guidelines [require](https://docs.zephyrproject.org/latest/contribute/guidelines.html#dco-sign-off) require to not remove existing sign-off.

So the second iteration, in this PR and what ended up being merged, I tried to convey the same thing you'd asked me to, by also taking into account your sign-off *had* to be there. So I put myself as author of the commit (arguably I did *some*, albeit a very very small portion, work on it, so it's hopefully not incorrect) as a way to not be in a situation where you'd be both author and sign-off (as you probably know, our compliance checks require that the git author appears in the list of sign-offs), as you'd explicitly told me you didn't want that.
So in the end it's "Author: Benjamin, Sign-off: Benjamin, Sign-off: Andrew". What's arguably missing is Co-authored-by: Andrew but 1/ like you mentioned, it's not really a git "native" thing, and in Zephyr sign-off typically implies (co-)authorship and 2/ it doesn't look like tools like Gitlens care about Co-authored-by anyway, although apparently GitHub does https://github.com/zephyrproject-rtos/zephyr/commit/e58dfac3742c4d7c675ebe92a0992619c957702d.

So yes you're right, perhaps it was rushed in that you would have in fact been happy with both being git author and sign-off entry in the squashed commit in the end (I'd reached to you about that, and I should have probably waited longer to hear back from you, as you'd maybe change your mind, so I'm sorry I didn't), or that I should have added Co-authored-by in addition to Sign-off even if it felt redundant to me -- looks like maybe it wasn't.

> Edit: I don't think there's any practical[1] way to change this after the fact. I do feel it's reasonable to recognise (for my self) and state (to others) my thoughts and feelings about this.

Yep, that's perfectly reasonable and understandable.

**Comment by kartben** [comment]
> I really don't think kartben meant to do that: they didn't do it in their previous version of the PR, but post-review feedback they did.

And to be extra clear: yes, I of course had no intention to come across as claiming authorship in any way, and would have been happy for my name to not even appear on the commits at all had it been an option :)

**Comment by carlescufi** [comment]
> This, maintainers (@soburi @carlescufi @nordicjm), is why I really really really tried so many times to discourage futzing around and rebasing code. It's such an easy mistake to fall into exactly this, where as far as git's history will record (and gitlens et al will show when people are reviewing this), all of the work in that commit was done by @kartben .

@ajf58 just a quick note here: the rebasing/squashing that was originally requested in your PR was to be done and signed off entirely by you, so at that point I don't think there was any risk of mismatching authorship in commits. Of course, the moment a second person got involved then this "issue" suddenly appeared and creative usage of Git commit author, signed-off-by and co-authored-by was required in order to try and reflect the reality of things in the commits.

**Comment by soburi** [comment]
> > the last chance to merge in your name.
> 
> @soburi , that's not what I'm talking about. I am talking about the difference between a committer and an author within `git`. Please take this as constructive feedback: quite often when I try and raise a question about one topic you confidently reply about something else, often at great length. This makes it quite challenging to move the conversation forward.
> 
> Please go and look at [5c39bb2](https://github.com/zephyrproject-rtos/zephyr/commit/5c39bb22a0caff1ace4d360a9e0f2f7066c55dfc) . In the process of squashing, both the author (which should be me), and the committer have changed. IMHO former is not correct, the latter is. I really don't think kartben meant to do that: they didn't do it in their previous version of the PR, but post-review feedback they did.
> 
> Edit: I don't think there's any practical[1] way to change this after the fact. I do feel it's reasonable to recognise (for my self) and state (to others) my thoughts and feelings about this.
> 
> [1] No one is going to force-push to `main` to change this, for many reasons, and I don't expect them to.

I apologize if that was an inappropriate comment.
I appreciate your point and advice.

**Pull request merged**
